File: modules/admin-ui/src/main/resources/web/admin/common/js/data/PropertyId.ts
module api.data {

    export class PropertyId implements api.Equitable {

        private value: string;

        constructor(value: string) {
            this.value = value;
        }

        public equals(o: Equitable): boolean {

            if (!api.ObjectHelper.iFrameSafeInstanceOf(o, PropertyId)) {
                return false;
            }

            var other = <PropertyId>o;

            if (!api.ObjectHelper.stringEquals(this.value, other.value)) {
                return false;
            }

            return true;
        }

        public toString(): string {
            return this.value;
        }
    }
}

File: modules/core-api/src/main/java/com/enonic/xp/core/data/PropertyId.java
package com.enonic.xp.core.data;

import java.util.Objects;

public final class PropertyId
{
    private final String value;

    PropertyId( final String value )
    {
        this.value = value;
    }

    @Override
    public boolean equals( final Object o )
    {
        if ( this == o )
        {
            return true;
        }
        if ( o == null || getClass() != o.getClass() )
        {
            return false;
        }

        final PropertyId other = (PropertyId) o;
        return Objects.equals( value, other.value );
    }

    @Override
    public int hashCode()
    {
        return Objects.hashCode( value );
    }

    @Override
    public String toString()
    {
        return this.value;
    }
}
